tests/src/LuaTests.cs

```diff
@@ -1766,6 +1766,111 @@ public void TestPrintChars()
             }
         }
 
+        [Test]
+        public void TestPrintNoParams()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.DoString(@"print()");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintSingleNil()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.DoString(@"print(nil)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNil()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.DoString(@"print(nil, nil, nil, nil, nil, nil)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNilFirstNumber()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.DoString(@"print(2, nil, nil, nil, nil, nil)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNilLastNumber()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.DoString(@"print(nil, nil, nil, nil, nil, 2)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintNoParamsParamsDetour()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.RegisterFunction("print", PrintOverrides.GetPrintDetourMethodInfo());
+                lua.DoString(@"print()");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintSingleNilParamsDetour()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.RegisterFunction("print", PrintOverrides.GetPrintDetourMethodInfo());
+                lua.DoString(@"print(nil)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNilParamsDetour()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.RegisterFunction("print", PrintOverrides.GetPrintDetourMethodInfo());
+                lua.DoString(@"print(nil, nil, nil, nil, nil, nil)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNilFirstNumberParamsDetour()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.RegisterFunction("print", PrintOverrides.GetPrintDetourMethodInfo());
+                lua.DoString(@"print(2, nil, nil, nil, nil, nil)");
+                Assert.IsTrue(true);
+            }
+        }
+
+        [Test]
+        public void TestPrintManyNilLastNumberParamsDetour()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua.RegisterFunction("print", PrintOverrides.GetPrintDetourMethodInfo());
```

Comment:
Also, I think there are easier ways to binding a method without all the reflection stuff, take a look in the other tests that we have.

Reply:
Thanks for the heads-up! After fooling around a bit more I realized the only reason I needed all the reflection stuff was because I mad the method private (which in my head was a great idea since only Lua would be using the method). For simplicity's sake I've changed it to public and simplified the registration